pkg/channel/testing/channel_mock.go

```diff
@@ -75,8 +76,15 @@ func (_m *MockAppChannel) HealthProbe(ctx context.Context) (bool, error) {
 	return r0, r1
 }
 
+func (_m *MockAppChannel) Init() {
+	_m.requestsReceived = make(map[string]*v1.InvokeMethodRequest)
+}
+
 // InvokeMethod provides a mock function with given fields: ctx, req
 func (_m *MockAppChannel) InvokeMethod(ctx context.Context, req *v1.InvokeMethodRequest) (*v1.InvokeMethodResponse, error) {
+	if _m.requestsReceived != nil {
```

Comment:
This is likely going to run into race conditions. Have you tried running `make test-race`?

Reply:
Using a lock. That should work fine when working with shared resource. Addressed under https://github.com/dapr/dapr/pull/5398.